### PR TITLE
Feat: add more builtin audits

### PR DESCRIPTION
**Pull request by z3z1ma**

### CODE REVIEW

File: sqlmesh/core/audit/builtin.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
 from __future__ import annotations
 
+import sqlglot.expressions as exp
```
Comment (tobymao) [inline sqlmesh/core/audit/builtin.py:3]:
```suggestion
from sqlglot import exp
```

File: sqlmesh/core/audit/builtin.py [hunk 2]
```diff
@@ -76,3 +83,238 @@
...
+    name="relationship",
+    defaults={"from_condition": exp.null(), "to_condition": exp.null()},
+    query="""
+SELECT
+  child.source_column
+FROM (
+  SELECT @source_column as source_column
+  FROM @this_model
+  WHERE NOT @source_column IS NULL
+  AND @IF(@from_condition IS NOT NULL, @from_condition, 1=1)
```
Comment (tobymao) [inline sqlmesh/core/audit/builtin.py:98]:
```suggestion
  AND @from_condition
```

File: sqlmesh/core/audit/builtin.py [hunk 3]
```diff
@@ -76,3 +83,238 @@
...
+) AS child
+LEFT JOIN (
+  SELECT @target_column AS target_column
+  FROM @to
+  @WHERE(@to_condition IS NOT NULL) @to_condition
```
Comment (tobymao) [inline sqlmesh/core/audit/builtin.py:103]:
```suggestion
  WHERE @to_condition 
```

File: sqlmesh/core/audit/builtin.py [hunk 4]
```diff
@@ -76,3 +83,238 @@
...
+# accepted_range(column=age, min_v=10)
+# accepted_range(column=age, max_v=50)
+accepted_range_audit = Audit(
+    name="accepted_range",
+    defaults={"min_v": exp.null(), "max_v": exp.null(), "inclusive": exp.true()},
+    query="""
+SELECT *
+FROM @this_model
+WHERE
+  1=2
```
Comment (tobymao) [inline sqlmesh/core/audit/builtin.py:120]:
```suggestion
  False
```

File: sqlmesh/core/audit/builtin.py [hunk 5]
```diff
@@ -76,3 +83,238 @@
...
+  OR @IF(@min_v IS NOT NULL AND @inclusive, @column <= @min_v, 1=2)
```
Comment (tobymao) [inline sqlmesh/core/audit/builtin.py:121]: same type of stuff here

File: sqlmesh/core/audit/builtin.py [hunk 6]
```diff
@@ -76,3 +83,238 @@
...
+  GROUP BY @target_column
+),
+except_a AS (
+  SELECT *
+  FROM table_a
+  EXCEPT
+  SELECT *
+  FROM table_b
+),
+except_b as (
```
Comment (tobymao) [inline sqlmesh/core/audit/builtin.py:229]: AS

File: sqlmesh/core/audit/builtin.py [hunk 7]
```diff
@@ -76,3 +83,238 @@
...
+""",
+)
+
+# mutually_exclusive(column=column_name, with=some.model)
+# TODO: make inclusivity configurable
+mutually_exclusive_ranges_audit = Audit(
+    name="mutually_exclusive_ranges",
+    defaults={"partition_clause": exp.null()},
+    query="""
+with window_functions as (
```
Comment (tobymao) [inline sqlmesh/core/audit/builtin.py:281]: run all the queries through sqlglot to fix casing and style

File: sqlmesh/core/audit/definition.py [hunk 8]
```diff
@@ -31,6 +31,8 @@ class AuditMeta(PydanticModel):
     """Setting this to `true` will cause this audit to be skipped. Defaults to `false`."""
     blocking: bool = True
     """Setting this to `true` will cause the pipeline execution to stop if this audit fails. Defaults to `true`."""
+    defaults: t.Dict[str, exp.Expression] = Field(default_factory=dict)
```
Comment (tobymao) [inline sqlmesh/core/audit/definition.py:34]: you don't need field, you can just do 

defaults = {}

File: sqlmesh/core/audit/definition.py [hunk 9]
```diff
@@ -250,3 +262,14 @@ class AuditResult(PydanticModel):
...
+
+
+def _maybe_parse_arg_pair(e: exp.Expression) -> t.Tuple[str, exp.Expression]:
+    if isinstance(e, exp.EQ):
+        k = e.left
+        if isinstance(k, exp.Column):
+            k = k.this
+        if not isinstance(k, (exp.Identifier, exp.Literal)):
+            raise_config_error(f"Invalid defaults key: {k}", error_type=AuditConfigError)
+        return str(k.this), e.right
```
Comment (tobymao) [inline sqlmesh/core/audit/definition.py:274]: return e.left.name, e.right 

this is all you need i think

File: sqlmesh/core/audit/builtin.py (hunk 5 above)
Comment (z3z1ma) [reply sqlmesh/core/audit/builtin.py:121]: we need to support min_v or max_v of 0 which would be falsey in python. So null makes a better sentinel value, no?

File: sqlmesh/core/audit/builtin.py [hunk 10]
```diff
@@ -26,21 +29,21 @@
   SELECT
     @EACH(
       @columns,
-      c -> row_number() OVER (PARTITION BY c ORDER BY 1) AS @SQL('@{c}_rank')
+      c -> row_number() OVER (PARTITION BY c ORDER BY 1) AS "@{c}_rank"
```
Comment (tobymao) [inline sqlmesh/core/audit/builtin.py:32]: you could also do as rank_@c now

**Review by tobymao — APPROVED** [review]

**Pull request merged**